### PR TITLE
Run Third party audit with forbidden APIs CLI  (part3/3)

### PR DESCRIPTION
**Pull request by alpar-t**
This concludes the work to switch to using runtime java for forbidden APIs check and should allow us to always run Gradle with compile java. 

The new implementation is functional  equivalent with the old, ant based one. 
It parses task standard error to get the missing classes and violations in the same way. 
I considered re-using `ForbiddenApisCliTask` but Gradle makes it hard to build inheritance with tasks that have task actions , since the order of the task actions can't be controlled.
This inheritance isn't dully desired either as the third party audit task is much more opinionated and we don't want to expose some of the configuration. 
We could probably extract a common base class without any task actions, but probably more trouble than it's worth.

### CODE REVIEW

File: buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ForbiddenApisCliTask.java [hunk 1]
```diff
@@ -57,14 +59,6 @@ public void setTargetCompatibility(JavaVersion targetCompatibility) {
         this.targetCompatibility = targetCompatibility;
     }
 
-    public Action<JavaExecSpec> getExecAction() {
```
Comment (atorok) [inline buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ForbiddenApisCliTask.java]: The action can't be an `@Input`, but we need to pick up changes in runtime java version/

File: plugins/ingest-attachment/build.gradle [hunk 2]
```diff
@@ -2106,7 +2106,27 @@ thirdPartyAudit.excludes = [
   'ucar.nc2.dataset.NetcdfDataset'
 ]
 
-if (JavaVersion.current() > JavaVersion.VERSION_1_8) {
+if (project.runtimeJavaVersion == JavaVersion.VERSION_1_8) {
+    thirdPartyAudit.excludes += [
+            // TODO: Why is this needed ?
+            'com.sun.javadoc.ClassDoc',
```
Comment (atorok) [inline plugins/ingest-attachment/build.gradle]: I'm a bit puzzled why this came up now. 
AFAIK this is part of the jdk, and not the jre ( javac ). Maybe the ant runner did something to load that.

File: server/build.gradle [hunk 3]
```diff
@@ -304,17 +304,22 @@ thirdPartyAudit.excludes = [
...
-    'java.io.ObjectInputFilter$Config',
-    'java.io.ObjectInputFilter$FilterInfo',
-    'java.io.ObjectInputFilter$Status'
+          // Used by Log4J 2.11.1
+            'java.io.ObjectInputFilter',
+            'java.io.ObjectInputFilter$Config',
+            'java.io.ObjectInputFilter$FilterInfo',
+            'java.io.ObjectInputFilter$Status',
+          // added in 9
+            'java.lang.ProcessHandle',
```
Comment (atorok) [inline server/build.gradle]: AFAIK this came up now as it's not in 8 for sure. 
Maybe Gradle had something on the cp for compatibility ?

**Comment by atorok** [comment]
@rjernst ready for review.

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**